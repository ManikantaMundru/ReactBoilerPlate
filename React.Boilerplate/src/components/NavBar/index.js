/**
 *
 * NavBar
 *
 */

import React, { memo } from 'react';
import PropTypes from 'prop-types';
import StyledNavBar from './styledComponents/styledNavBar';
import Button from '../Button';

const NavBar = props => {
  const { isAuthenticated, loginWithRedirect, logout, user } = props.auth;

  return (
    <div>
      {!isAuthenticated && (
        <StyledNavBar>
          <h1>Please click login to get started</h1>
          <Button onClick={() => loginWithRedirect({})} title="Log in" />
        </StyledNavBar>
      )}

      {isAuthenticated && (
        <StyledNavBar>
          <h1>Welcome {user.name}, you are now logged in</h1>
          <img src={user.picture} alt="avatar" />
          <Button onClick={() => logout()} title="Log out" />
        </StyledNavBar>
      )}
    </div>
  );
};

NavBar.propTypes = {
  auth: PropTypes.object,
};

export default memo(NavBar);
