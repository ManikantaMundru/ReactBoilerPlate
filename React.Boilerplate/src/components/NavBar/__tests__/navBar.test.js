import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import NavBar from '../index';

afterEach(cleanup);

test('does component render', () => {
  const { asFragment } = render(
    <NavBar
      auth={{
        isAuthenticated: false,
        loginWithRedirect: () => {
          alert('login'); //eslint-disable-line
        },
        logout: () => {
          alert('logout'); //eslint-disable-line
        },
        user: {
          nickname: 'tester',
          picture:
            'https://images.squarespace-cdn.com/content/5c4f53218f5130618631fc97/1550519005127-SXL0XBRPDP4NJ762XBRV/RapidDeploy+logo+2019+TM-01.png?format=1500w&content-type=image%2Fpng',
        },
      }}
    />,
  );
  expect(asFragment()).toMatchSnapshot();
});

test('does component render', () => {
  const { asFragment } = render(
    <NavBar
      auth={{
        isAuthenticated: true,
        loginWithRedirect: () => {
          alert('login'); //eslint-disable-line
        },
        logout: () => {
          alert('logout'); //eslint-disable-line
        },
        user: {
          nickname: 'tester',
          picture:
            'https://images.squarespace-cdn.com/content/5c4f53218f5130618631fc97/1550519005127-SXL0XBRPDP4NJ762XBRV/RapidDeploy+logo+2019+TM-01.png?format=1500w&content-type=image%2Fpng',
        },
      }}
    />,
  );
  expect(asFragment()).toMatchSnapshot();
});
