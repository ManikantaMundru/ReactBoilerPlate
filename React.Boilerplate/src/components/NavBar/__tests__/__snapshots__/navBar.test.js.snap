// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`does component render 1`] = `
<DocumentFragment>
  <div>
    <div
      class="sc-bdVaJa bDWFJH"
    >
      <h1>
        Please click login to get started
      </h1>
      <button
        class="sc-bwzfXH bvjjuq"
      >
        Log in
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`does component render 2`] = `
<DocumentFragment>
  <div>
    <div
      class="sc-bdVaJa bDWFJH"
    >
      <h1>
        Welcome tester, you are now logged in
      </h1>
      <img
        alt="avatar"
        src="https://images.squarespace-cdn.com/content/5c4f53218f5130618631fc97/1550519005127-SXL0XBRPDP4NJ762XBRV/RapidDeploy+logo+2019+TM-01.png?format=1500w&content-type=image%2Fpng"
      />
      <button
        class="sc-bwzfXH bvjjuq"
      >
        Log out
      </button>
    </div>
  </div>
</DocumentFragment>
`;
